### ISSUE TITLE
Fails to install in a fresh/new virtual environment

### ISSUE THREAD

**Issue by mcgillowen** [wwkimball/yamlpath#92]
## Operating System

1. Name/Distribution: KDE Neon (Ubuntu)
2. Version: 20.04

## Version of Python and packages in use at the time of the issue.

1. [Distribution](https://wiki.python.org/moin/PythonDistributions): CPython installed using asdf-vm
2. Python Version: 3.8.5
3. Version of yamlpath installed: N/A, trying to install 3.x
4. Version of ruamel.yaml installed: N/A

## Complete steps to reproduce the issue when triggered via:

1. Install `yamlpath` in a fresh/new virtual environment with the following commands

```
$ python3 -m venv .venv
$ source .venv/bin/activate
$ pip install yamlpath
```

## Expected Outcome

`yamlpath` gets installed successfully

## Actual Outcome

Install fails with the following error:

```
Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-pddralma/yamlpath/setup.py", line 3, in <module>
        from yamlpath.common import YAMLPATH_VERSION
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/__init__.py", line 2, in <module>
        from yamlpath.yamlpath import YAMLPath
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/yamlpath.py", line 9, in <module>
        from yamlpath.types import PathSegment
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/types/__init__.py", line 2, in <module>
        from .pathattributes import PathAttributes
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/types/pathattributes.py", line 4, in <module>
        from yamlpath.path import CollectorTerms
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/path/__init__.py", line 2, in <module>
        from .collectorterms import CollectorTerms
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/path/collectorterms.py", line 6, in <module>
        from yamlpath.enums import CollectorOperators
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/enums/__init__.py", line 8, in <module>
        from .yamlvalueformats import YAMLValueFormats
      File "/tmp/pip-install-pddralma/yamlpath/yamlpath/enums/yamlvalueformats.py", line 9, in <module>
        from ruamel.yaml.scalarstring import (
    ModuleNotFoundError: No module named 'ruamel'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

The error does not occur if `ruamel.yaml` is installed first using `pip install ruamel.yaml`.

My guess is that this comes from the following import in `setup.py`

```
from yamlpath.common import YAMLPATH_VERSION
```

Since the imports are resolved before setup_tools installs the packages listed in `install_requires`

**Comment by wwkimball** [comment]
Good catch!  I'm surprised this doesn't show up in any of my automated tests; I'll have to add a deliberate case for this.  I'd followed a so-called "Best Practice" for centralizing the version number of releases so as to avoid having to update so many individual files.  The author of the guide neglected to mention this as a side-effect when depending on other packages, too.  I'll get this fixed ASAP.

Since I'm very close to releasing the next version with some important enhancements, I'll roll this fix forward into the next release.

**Comment by wwkimball** [comment]
Issues with TravisCI are delaying the release with this fix.  I'm posting progress at PR #94.

**Comment by wwkimball** [comment]
I have (finally) released 3.3.0, which fixes this issue.  Thank you for reporting it!
